Add methods with f64 std equivelant to Math object

Nickforall opened the pull request:
I implemented some more Math functions which already have an equivelant in the rust standard library. There are still a couple more that have to be implemented, but they require a bit more work as they have some side effects or are not in the rust standard library. Those are out of scope for this PR.

<details> 
  <summary>List of Math object functions</summary>

- [x]  [Math.abs()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/abs)
- [x]  [Math.acos()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/acos)
- [x]  [Math.acosh()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/acosh)
- [x]  [Math.asin()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/asin)
- [x]  [Math.asinh()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/asinh)
- [x]  [Math.atan()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/atan)
- [x]  [Math.atan2()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/atan2)
- [x]  [Math.atanh()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/atanh)
- [x]  [Math.cbrt()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/cbrt)
- [x]  [Math.ceil()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/ceil)
- [ ]  [Math.clz32()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/clz32)
- [x]  [Math.cos()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/cos)
- [x]  [Math.cosh()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/cosh)
- [x]  [Math.exp()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/exp)
- [ ]  [Math.expm1()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/expm1)
- [x]  [Math.floor()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/floor)
- [ ]  [Math.fround()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/fround)
- [ ]  [Math.hypot()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/hypot)
- [ ]  [Math.imul()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/imul)
- [x]  [Math.log()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/log)
- [x]  [Math.log10()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/log10)
- [x]  [Math.log1p()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/log1p)
- [x]  [Math.log2()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/log2)
- [x]  [Math.max()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/max)
- [x]  [Math.min()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/min)
- [x]  [Math.pow()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/pow)
- [x]  [Math.random()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random)
- [x]  [Math.round()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/round)
- [x]  [Math.sign()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/sign)
- [x]  [Math.sin()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/sin)
- [x]  [Math.sinh()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/sinh)
- [x]  [Math.sqrt()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/sqrt)
- [x]  [Math.tan()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/tan)
- [x]  [Math.tanh()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/tanh)
- [x]  [Math.trunc()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/trunc)

</details>

jasonwilliams commented:
Hey this is awesome, thanks!
will take a look (sorry ive been a while)

jasonwilliams commented:
@Nickforall looks ok, are you planning to add tests for these?

jasonwilliams commented:
Some errors
```js
Math.acosh(0.999999999999);
// expected output: NaN
```

```
ParsingError: Expected token 'identifier', got '999999999999' in routine 'field access' at line 1, col 14
error: process didn't exit successfully: `target\debug\boa_cli.exe` (exit code: 1)
The terminal process terminated with exit code: 1
```

jasonwilliams commented:
decimal number parsing is a bit messed up
Token output for `Math.atanh(0.5);`

```
[
    Token {
        data: Identifier(      
            "Math",
        ),
        pos: Position {        
            column_number: 1,  
            line_number: 1,    
        },
    },
    Token {
        data: Punctuator(      
            Dot,
        ),
        pos: Position {        
            column_number: 5,  
            line_number: 1,    
        },
    },
    Token {
        data: Identifier(      
            "atanh",
        ),
        pos: Position {        
            column_number: 6,  
            line_number: 1,    
        },
    },
    Token {
        data: Punctuator(      
            OpenParen,
        ),
        pos: Position {        
            column_number: 11,
            line_number: 1,   
        },
    },
    Token {
        data: NumericLiteral( 
            0.0,
        ),
        pos: Position {       
            column_number: 12,
            line_number: 1,   
        },
    },
    Token {
        data: Punctuator(     
            Dot,
        ),
        pos: Position {       
            column_number: 13,
            line_number: 1,   
        },
    },
    Token {
        data: NumericLiteral( 
            5.0,
        ),
        pos: Position {       
            column_number: 14,
            line_number: 1,   
        },
    },
    Token {
        data: Punctuator(     
            CloseParen,       
        ),
        pos: Position {       
            column_number: 15,
            line_number: 1,   
        },
    },
    Token {
        data: Punctuator(
            Semicolon,
        ),
        pos: Position {
            column_number: 16,
            line_number: 1,
        },
    },
]
```

Nickforall commented:
Yeah sure, I’ll add some tests when i have time later this week :)

On Fri, 6 Mar 2020 at 22:10, Jason Williams <notifications@github.com>
wrote:

> @Nickforall <https://github.com/Nickforall> looks ok, are you planning to
> add tests for these?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/jasonwilliams/boa/pull/260?email_source=notifications&email_token=ABGBWMUDS4GFGOVT6LIJA63RGFRFHA5CNFSM4K2VMZCKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEOC3WYQ#issuecomment-595966818>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABGBWMSX4JFQ7WH7FJEECCDRGFRFHANCNFSM4K2VMZCA>
> .
>

jasonwilliams commented:
We have mini-test suite for lexing numbers, looks like 0.1 should be added:
https://github.com/jasonwilliams/boa/blob/master/boa/src/syntax/lexer/tests.rs#L384-L405

Nickforall commented:
It's the octal parsing (for example, with a leading zero, numbers are parsed as octal: `0777 => 511`). I kind of don't want to mess around with the full number parsing arm, but it seems like it's necessary to do this elegantly, so I will give this a shot in another PR :)

You can merge this one after. Whatever works for you :)

Nickforall commented:
Will rebase once #260 is merged.

Nickforall commented:
Rebased and added test for all methods. The tests are largely based on the [docs from Mozilla](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math)

jasonwilliams commented:
Thanks for your contribution @Nickforall it looks good
